### ISSUE TITLE
[Bug] Not possible to put cart in slot 1 and extension in slot 2 [sf#272]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/openMSX#573]
**Reported by manuelbi on 2007-08-04 12:13 UTC**
Example:

$ openmsx -carta somerom.rom -ext scc
info: Using default machine: Boosted_MSX2_EN
Fatal error: slot-a already in use.

Likewise, when using -cart, the cartridge ends up in slot 2 and the extension in slot 1.

This probably got introduced here:
2006-02-03 Wouter Vermaelen 
* Removed 'slot reservation' mechanism in command line parser:
- The order of -cart and -ext command line options is now important

A workaround is to use the 'ext' console command after startup.
Quite an annoying bug, e.g. if you want to test a Konami ROM adaptation for SCC in slot 2.

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2014-02-20 16:14 UTC**
This is actually introduced with this change:

commit 06f21cfc422207bbce725269d8d154eafe31fcb5
Author: Wouter Vermaelen m9710797@users.sourceforge.net
Date:   Fri Feb 17 20:58:55 2006 +0000

```
fixed inserting disk in external drive via command line
   * Tweaked order of command line parsing:
```
-          extensions should come before filename parsing to handle stuff like
-            openmsx -machine msx1 -ext fdd mydisk.dsk

This causes the 'ext' commandline option to get parsed always before the cart(a) option. In this case, it means the 'ext' option gets slot A and the conflict with 'carta' is clear.

However, I don't see a reason why extension should be parsed first. In fact, if I remove the PHASE_EXT stuff from ExtensionCli, the example from the commit message above still works in the current HEAD code.

So, I propose to remove the PHASE_EXT stuff so you can use the commandline order to influence this.

More generally, I'm preparing a patch to enable 'exta, extb' etc. options to offer the user even more control on where extensions end up.

**Comment by openMSX-import** [comment]
**Updated by manuelbi on 2014-02-20 17:06 UTC**
- **status**: open --> closed-fixed
- **assigned_to**: Wouter Vermaelen --> Manuel Bilderbeek
- **Group**:  --> Next_release

**Comment by openMSX-import** [comment]
**Commented by manuelbi on 2014-02-20 17:06 UTC**
This has now been fixed with commit e6a51b90b40e6614bad671074220b62206b8eb72:

Author: Manuel Bilderbeek Manuel.Bilderbeek@gmail.com
Date:   Thu Feb 20 18:01:04 2014 +0100

```
Do not parse -ext earlier than "other" options on the cmdline.

Because -ext was parsed before -carta, using -ext and -carta didn't work
as expected: slot A was taken by the extension and the cartridge fails
to insert. This is now working, so you have more control which thing
ends up in which slot.
```

More fine control is for a later patch. :)
